lib/semian/protected_resource.rb

```diff
@@ -2,35 +2,59 @@ module Semian
   class ProtectedResource
     extend Forwardable
 
-    def_delegators :@resource, :destroy, :count, :semid, :tickets, :name
+    def_delegators :@bulkhead, :destroy, :count, :semid, :tickets, :name
```

Comment:
I personally like this. I found this confusing too. The only issue is that in the C code, it's still referenced as `Resource`. It seems unnecessary to change both though.